Rx.NET/Documentation/adr/0001-net7.0-era-tooling-update.md

```diff
@@ -0,0 +1,140 @@
+
+### Windows SDK version
+
+In any projects that target `uap10.0.16299`, change that to `uap10.0.18362`. (The goal with this particular set of changes is to get everything building on modern tooling while changing as little as possible. This is the smallest version change that meets those goals, and that's why we're not updating to a more recent version.)
+
+Change the `Tests.System.Reactive.Uwp.DeviceRunner` test runner project to target SDK 10.0.18362.
+
+### New analyzer diagnostics
+
+New diagnostic messages caused by more aggressive analyzers in the current SDK will be dealt with in two phases
+
+1. to enable preview builds of the next version to be made available as early as possible, we will initially disable the new diagnostics
+2. once preview builds have been released, we will update the codebase in cases where the new warnings make suggestions we want to take, and add suitable configuration to prevent warnings suggesting changes we do not want
+
+### UWP test runner issues
+
+It Coverlet-related build problems can be dealt with by setting the `NETCoreSdkVersion` build variable in the UWP test project.
+
+As for the lack of current xUnit support, it looks like the only realistic option at this point is not to use xUnit. An experimental spike has been done to demonstrate that moving to the MSTest framework is relatively straightforward. A handful of tests fail because they turned out to be dependent on xUnit supplying a non-null `SynchronizationContext`. These tests would need to be examined—it is possible the failures represent a real problem that needs addressing, but if not we either need to modify the tests so that their expectations are in line with the absence of a `SynchronizationContext`, or to arrange for a `SynchronizationContext` to be set up when those tests run (or possibly do both, so that the relevant tests run both with and without a `SynchronizationContext`).
+
+## Consequences
+
+These changes will have the following positive effects:
+
+1. it will be possible to download the repository and, using current tooling (VS 2022 and .NET 7.0 SDK), open, build, and test `System.Reactive.sln` without errors, warnings or other diagnostic messages
+2. automated builds will now be running tests on .NET 6.0 and .NET 7.0 (in addition to .NET Framework)
+3. when we release components based on these changes to NuGet, they will no longer target frameworks that are out of support, and will include `net6.0` targets
+4. if setting `<IsTrimmable>true</IsTrimmable>` turns out to be all that is required for trimmability, this will address that requirement
+
+It will impose the following new requirements for anyone wanting to build the solution:
+
+1. the repository must be cloned into a location where the path length is no longer than 25 characters
+1. Visual Studio 2022 must be installed
```

Comment:
Do we require VS2022 to build? Or do we require VS2022 if VS is used?

Reply:
I don't know how you install the parts that know how to build old-school UWP project files (notably the UWP test runner project) without installing Visual Studio. There are various MS build `.props` and `.targets` files required by that which don't get installed as part of the .NET SDK.

The status quo is that Visual Studio has always been stated as a requirement. It's the first instruction in https://github.com/dotnet/reactive/wiki/Building%20Testing%20and%20Debugging

So this is only a change of version, which seems reasonable for this initial "update all the tooling while changing as little as possible" phase.

It might be that there is some SDK we can install that enables this to be built without VS being installed, but it would require some research to work out what is and isn't possible, and I'm not sure this is the right time for that. (I think V.next.next would be a more appropriate milestone.)